### PR TITLE
📦 Add link check action

### PR DESCRIPTION
**Pull request by TechWiz-3**
In this PR, I have added a workflow which checks links and fails if any of them are broken.

Disclaimer: this action is my own action, the repo is [here](https://github.com/TechWiz-3/linksafe)

You can view the logs of the workflow on my fork [here](https://github.com/TechWiz-3/awesome-devops/runs/7597938097?check_suite_focus=true). 2 links were found to be broken.

Let me know if you'd like me to change anything, any feedback is appreciated either way :)

### CODE REVIEW

**Comment by wmariuss** [comment]
Hi,

Nice workflow. The question is can we make also If the URL is safe enought?

**Comment by TechWiz-3** [comment]
Thank you 😌 

What do you mean safe enough though? If the connection is not secure (i.e. no SSL certificate) the link is considered to be broken and is added to the bad links list, I could also create an option that enforces only https links too if that's what you mean 🤔

**Comment by wmariuss** [comment]
Yes, if it has SSL cert valid, but also, the content is safe.

**Comment by TechWiz-3** [comment]
Surely it wouldn't be added if the content isn't safe?

In any case, that would be a cool feature to explore, any ideas on how that can be measure? Also, any ideas on what defines 'safe' content?

**Comment by wmariuss** [comment]
Ok, add SSL validation for now. and in the future is good to have content validation as well.

**Comment by TechWiz-3** [comment]
SSL validation is already in place
Screen Shot 2022-08-13 at 12 19 16 am

File: .github/workflows/linkcheck.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Link-check
+on: [push]
```
Comment (wmariuss) [inline .github/workflows/linkcheck.yml:2]: Is is possible to put this:
`workflow_dispatch:` instead push?
I want to be manually for now.

File: .github/workflows/linkcheck.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run linksafe
+        uses: TechWiz-3/linksafe@main
```
Comment (wmariuss) [inline .github/workflows/linkcheck.yml:10]: Can you have a tag version instead main branch?

File: .github/workflows/linkcheck.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        with:
+          dirs: ".,./docs"
+          verbose: true
+          whitelist_files: "./CONTRIBUTING.md"
+          whitelist_links: "https://ceph.io/,https://mattermost.com/,https://tyk.io/,https://konghq.com/,https://konghq.com/products/kong-gateway/kong-proxy,https://www.saltstack.com/,https://www.docker.com/,https://www.gluster.org/,https://cloud.vmware.com/,https://www.vultr.com/,https://www.linkedin.com/in/wmariuss"
```
Comment (wmariuss) [inline .github/workflows/linkcheck.yml:15]: Why is this list? Should not scan all?

**Comment by wmariuss** [comment]
> Any updates @wmariuss ?

Please see the comments.

File: .github/workflows/linkcheck.yml (hunk 3 above)
Comment (TechWiz-3) [reply .github/workflows/linkcheck.yml:15]: As the link checker uses HTTP response codes to determine whether a link is good or not, some links will come with a status code that *could* be broken but could also be OK. When this happens and you know a link is good, you don't want the checker flagging them in future, so you add them to the whitelist

**Pull request closed without merging**